sound/soc/sof/pm.c

```diff
@@ -435,12 +435,66 @@ EXPORT_SYMBOL(snd_sof_set_d0_substate);
 
 int snd_sof_resume(struct device *dev)
 {
+	struct snd_sof_dev *sdev = dev_get_drvdata(dev);
+	struct pci_dev *pci = to_pci_dev(sdev->dev);
+	int ret;
+
+	if (sdev->s0_suspend && sof_ops(sdev)->set_power_state) {
+		/* resuming from D0I3 */
+		dev_dbg(sdev->dev, "ADSP will exit from D0i3...\n");
+		ret = snd_sof_set_d0_substate(sdev, SOF_DSP_D0I0);
+		if (ret < 0) {
+			dev_err(sdev->dev, "error: failed to exit from D0I3 %d\n",
+				ret);
+			return ret;
+		}
+
+		pci_restore_state(pci);
+		disable_irq_wake(pci->irq);
```

Comment:
oh, this is not good at all. You CANNOT put PCI-related stuff in the PM core.

Reply:
OK, changing this.